src/webapi/html_elements/slot.rs

```diff
@@ -0,0 +1,77 @@
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webapi::event_target::{IEventTarget, EventTarget};
+use webapi::node::{INode, Node};
+use webapi::element::{IElement, Element};
+use webapi::html_element::{IHtmlElement, HtmlElement};
+
+/// The HTML `<slot>` element represents a placeholder inside a web component that
+/// you can fill with your own markup, which lets you create separate DOM trees and
+/// present them together.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot)
+// https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "HTMLSlotElement")]
+#[reference(subclass_of(EventTarget, Node, Element, HtmlElement))]
+pub struct SlotElement( Reference );
+
+impl IEventTarget for SlotElement {}
+impl INode for SlotElement {}
+impl IElement for SlotElement {}
+impl IHtmlElement for SlotElement {}
+
+impl SlotElement {
+    /// The slot's name
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/name)
+    // https://html.spec.whatwg.org/multipage/scripting.html#attr-slot-name
+    #[inline]
+    pub fn name ( &self ) -> String {
+        js! (
+            return @{self}.name;
+        ).try_into().unwrap()
+    }
+
+    /// Setter of name.
+    #[inline]
+    pub fn set_name<S: AsRef<str>>( &self, new_name: S ) {
```

Comment:
Is there any particular reason why you made it `S: AsRef<str>`? In general we usually take `&str` and have the user convert it themselves.

Reply:
No particular reason, I just feel it would be easier to use.